opal/clearwater/component.rb

```diff
@@ -118,6 +118,129 @@ module Component
+      auto_play: 'autoPlay',
+      capture: 'capture',
+      cell_padding: 'cellPadding',
+      cell_spacing: 'cellSpacing',
+      challenge: 'challenge',
+      char_set: 'charSet',
+      checked: 'checked',
+      class_id: 'classID',
+      class_name: 'className',
+      col_span: 'colSpan',
+      cols: 'cols',
+      content: 'content',
+      content_editable: 'contentEditable',
+      context_menu: 'contextMenu',
+      controls: 'controls',
+      coords: 'coords',
+      cross_origin: 'crossOrigin',
+      data: 'data',
+      date_time: 'dateTime',
+      default: 'default',
+      defer: 'defer',
+      dir: 'dir',
+      disabled: 'disabled',
+      download: 'download',
+      draggable: 'draggable',
+      enc_type: 'encType',
+      form: 'form',
+      form_action: 'formAction',
+      form_enc_type: 'formEncType',
+      form_method: 'formMethod',
+      form_no_validate: 'formNoValidate',
+      form_target: 'formTarget',
+      frame_border: 'frameBorder',
+      headers: 'headers',
+      height: 'height',
+      hidden: 'hidden',
+      high: 'high',
+      href: 'href',
+      href_lang: 'hrefLang',
+      html_for: 'htmlFor',
+      http_equiv: 'httpEquiv',
+      icon: 'icon',
+      id: 'id',
+      input_mode: 'inputMode',
+      integrity: 'integrity',
+      is: 'is',
+      key_params: 'keyParams',
+      key_type: 'keyType',
+      kind: 'kind',
+      label: 'label',
+      lang: 'lang',
+      list: 'list',
+      loop: 'loop',
+      low: 'low',
+      manifest: 'manifest',
+      margin_height: 'marginHeight',
+      margin_width: 'marginWidth',
+      max: 'max',
+      max_length: 'maxLength',
+      media: 'media',
+      media_group: 'mediaGroup',
+      method: 'method',
+      min: 'min',
+      min_length: 'minLength',
+      multiple: 'multiple',
+      muted: 'muted',
+      name: 'name',
+      no_validate: 'noValidate',
+      nonce: 'nonce',
+      open: 'open',
+      optimum: 'optimum',
+      pattern: 'pattern',
+      placeholder: 'placeholder',
+      poster: 'poster',
+      preload: 'preload',
+      radio_group: 'radioGroup',
+      read_only: 'readOnly',
+      rel: 'rel',
+      required: 'required',
+      reversed: 'reversed',
+      row_span: 'rowSpan',
+      rows: 'rows',
+      sandbox: 'sandbox',
+      scope: 'scope',
+      scoped: 'scoped',
+      scrolling: 'scrolling',
+      seamless: 'seamless',
+      selected: 'selected',
+      shape: 'shape',
+      size: 'size',
+      sizes: 'sizes',
+      span: 'span',
+      spell_check: 'spellCheck',
+      src: 'src',
+      src_doc: 'srcDoc',
+      src_lang: 'srcLang',
+      src_set: 'srcSet',
+      start: 'start',
+      step: 'step',
+      style: 'style',
+      summary: 'summary',
+      tab_index: 'tabIndex',
+      target: 'target',
+      title: 'title',
+      type: 'type',
+      use_map: 'useMap',
+      value: 'value',
+      width: 'width',
+      wmode: 'wmode',
+      wrap: 'wrap',
+    }
+
```

Comment:
I think the idea of filtering attributes through a hash *might* be really good for performance over the regex we use currently, but I would prefer to let anything that isn't modified pass through rather than having to whitelist all attributes so it doesn't become a maintenance burden.

However, I'm not sure camelCasing belongs at the component level. Since we provide a Ruby interface to `virtual-dom`, I think all translation to JS (including camelCase attributes) should happen there.